app/services/create_promotion_batch.rb

```diff
@@ -0,0 +1,7 @@
+class CreatePromotionBatch
```

Comment:
As I've mentioned before, this should live in Spree::Core, not in Spree::Backend - as it's not strictly related to the admin panel. Otherwise we won't be able to reuse this logic e.g. in the API.